nemo_skills/dataset/math-autoform/prepare.py

```diff
@@ -0,0 +1,192 @@
+        else:
+            return results
+
+
+def extract_attributes_from_name(file_name):
+    """Extract attributes from file path."""
+    eval_set, problem_type, fileid = file_name.split("/")[1:]
+    fileid = fileid.split(".")[0]
+    return eval_set, problem_type, fileid
+
+
+def extract_answer_string_2(answer_str):
+    """For two cases, inside the boxed expression, we needed a second iteration of parsing."""
+    left_string = "\\boxed"
+    idx = answer_str.rfind(left_string)
+
+    stripped_answer = answer_str[idx + len(left_string) :]
+    right_idx = stripped_answer.rfind("$")
+
+    stripped_answer = stripped_answer[:right_idx]
+    return stripped_answer
+
+
+def save_data(split, random_seed, validation_size):
+    output_dir = Path(__file__).absolute().parent
+    output_dir.mkdir(exist_ok=True)
+    actual_split = "test" if split == "test" else "train"
+    with tempfile.TemporaryDirectory() as temp_dir:
+        archive_filename = os.path.join(temp_dir, "temp.tar")
+        urllib.request.urlretrieve(DOWNLOAD_LINK, archive_filename)
+
+        split_instances_dict = defaultdict(list)
+
+        with tarfile.TarFile(archive_filename, mode="r") as reader_f:
+            for tar_member in reader_f:
+                filename = tar_member.name
+                if not filename.endswith(".json"):
+                    continue
+
+                eval_set, problem_type, fileid = extract_attributes_from_name(filename)
+                # TODO: we should just process all at ones, not do duplicate computation
+                if eval_set != actual_split:
+                    continue
+
+                content = json.loads(reader_f.extractfile(tar_member).read())
+                content["new_id"] = f"{eval_set}/{problem_type}/{fileid}.json"
+                # Load the solution with our identified fixes
+                content["reference_solution"] = _fix_solution(content["new_id"], content["solution"])
+                del content["solution"]
+
+                entries = find_boxed_entries(content["reference_solution"])
+                if len(entries) == 1:
+                    parsed_answer = entries[0]
+                if len(entries) > 1:
+                    parsed_answer = _post_fix_multi_answer(content["new_id"], entries)
+
+                if not (
+                    ("Find the equation" in content["problem"])
+                    or ("Enter the equation" in content["problem"])
+                    or ("What is the equation") in content["problem"]
+                    or ("described by the equation") in content["problem"]
+                    or ("Find an equation") in content["problem"]
+                ) and ("=" in parsed_answer):
+                    if parsed_answer.count("=") == 1:
+                        # For greater count, it means we're just predicting values of multiple variables
+                        parsed_answer = parsed_answer.split("=")[1]
+                content["predicted_answer"] = parsed_answer
```

Comment:
Is this the only change? If so, let's just have another config that would use expected_answer instead of predicted_answer and remove the new dataset. Useful to have that anyway as you might want to extend it to more than just math, and it's not scalable to duplicate every single dataset

Reply:
Also the id field is renamed as new_id. Since not all problems have id, model is usually trying to generate theorem name out of problem, by giving the prompt and id we instruct him to generate the name based on id.

Reply:
why does the name matter? Can we always use a fixed name "user_theorem" or something similar?

Reply:
I was using dataset mainly for one-click pipeline. Removed